extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -129,18 +152,50 @@ class AmpStickyAd extends AMP.BaseElement {
         const borderBottom = this.element./*OK*/offsetHeight;
         this.viewport_.updatePaddingBottom(borderBottom);
         this.addCloseButton_();
-        timer.delay(() => {
-          // Unfortunately we don't really have a good way to measure how long it
-          // takes to load an ad, so we'll just pretend it takes 1 second for
-          // now.
-          this.vsync_.mutate(() => {
-            this.element.classList.add('amp-sticky-ad-loaded');
-          });
-        }, 1000);
+        this.scheduleLayoutForAd_();
+      });
+    }
+  }
+
+  scheduleLayoutForAd_() {
+    if (this.ad_.isBuilt()) {
+      this.layoutAd_();
+    } else {
+      listenOnce(this.ad_, 'amp:built', () => {
```

Comment:
I've modified the function to two `dispatchCustomEventForTesting` and `dispatchCustomEvent`. By calling `dispatchCustomEvent` the message should be sent all the time